core/valor_core/managers.py

```diff
@@ -0,0 +1,322 @@
+import time
+from dataclasses import dataclass, field
+
+import pandas as pd
+from valor_core import enums, schemas
+from valor_core.detection import (
+    compute_detection_metrics,
+    create_detection_evaluation_inputs,
+)
+
+
+def _concatenate_df_if_not_empty(
+    df1: pd.DataFrame, df2: pd.DataFrame | None
+) -> pd.DataFrame:
+    """Checks to see if a dataframe is None before attempting a concatenation. Handles pandas warning about not using pd.concat on empty dataframes."""
+
+    if not df1.empty and (df2 is not None):
+        df1 = pd.concat(
+            [df1, df2],
+            ignore_index=True,
+        )
+    elif df1.empty and (df2 is not None):
+        df1 = df2
+
+    return df1
+
+
+@dataclass
+class ValorDetectionManager:
+    """
+    Manages the evaluation of object detection predictions against groundtruths.
+
+    Attributes
+    ----------
+    datum_uids : set
+        A set of unique identifiers for the data samples.
+    label_map : dict
```

Comment:
Could you update this to be `dict[schemas.Label, schemas.Label]`

Reply:
And datum_uids to `set[str]`

Reply:
And convert_annotations_to_type to `AnnotationType, optional` to match Numpy style